tests/zquantum/core/circuits/_serde_test.py

```diff
@@ -212,3 +219,11 @@ def test_load_save_circuitset_roundtrip(self):
         save_circuitset(circuitset, buf)
         buf.seek(0)
         assert load_circuitset(buf) == circuitset
+
+
+def test_make_symbols_map():
+    symbols_map = _make_symbols_map(["x", "y[2]", "y[3]"])
+    assert symbols_map == {
+        "x": sympy.Symbol("x"),
+        "y": {2: sympy.Symbol("y[2]"), 3: sympy.Symbol("y[3]")},
+    }
```

Comment:
**question**
`_make_symbols_map()` is used to generate `locals=` argument to [sympy.sympify](https://docs.sympy.org/latest/modules/core.html). Are we sure/was it tested that sympify() works with nested dictionaries?

`_make_symbols_map()` is a private function. WDYT about writing a unit test for `deserialize_expr()` instead? This would make the answer to my question above clearer + we we would only touch non-private stuff.

Reply:
Yup, it is tested indirectly through a [new item](https://github.com/zapatacomputing/z-quantum-core/blob/6eb5b957d400494af29fd26d47ad93f6b10ac8dd/tests/zquantum/core/circuits/_serde_test.py#L123-L127) added to `EXAMPLE_CIRCUITS` in the serde test.

Do you think that will be sufficient? Also I just deleted the test for _make_symbols_map in 84f6a684646d3b4042d8c6e1837e6c089f253f8d

Reply:
Oh, right. Makes sense :)